pallets/msa/src/lib.rs

```diff
@@ -877,6 +877,35 @@ pub mod pallet {
 			});
 			Ok(())
 		}
+
+		/// A generic endpoint to replay any offchain related event to fix any potential issues
+		#[pallet::call_index(13)]
+		#[pallet::weight(T::WeightInfo::reindex_offchain())]
+		pub fn reindex_offchain(
+			origin: OriginFor<T>,
+			event: OffchainReplayEvent<T>,
```

Comment:
A more generic event that can support different kind of offchain events and pallets. This extrinsic should be in some pallet and since we have the bulk of offchain implemented in this pallet it makes sense to keep it here. Even though it should be able to replay any offchain related to other pallets if required.